hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1135,6 +1135,12 @@ public AbfsInputStream openFileForRead(final Path path,
                     AzureServiceErrorCode.PATH_NOT_FOUND.getErrorCode(),
                     "openFileForRead must be used with files and not directories",
                     null);
+          } else {
```

Comment:
Special handling is not needed here

Reply:
This was leading to a few test failures otherwise, that depend on the 404 exception to be thrown

Reply:
We can directly throw e

Reply:
Taken.